### PR TITLE
Add Organization and Groups info to index file

### LINKED ISSUE #32: Consider exposing the group/organization that owns the spec
See discussion in https://github.com/w3c/browser-specs/issues/25#issuecomment-609054254.

### PR DESCRIPTION
**Pull request by tidoust**
This fetches some info about the organization and groups that own each spec, adding an `organization` property and a `groups` property.

Info for WHATWG, TC39, and WebGL specs is hardcoded. Info for W3C specs gets fetched from the W3C API for /TR specs and from the w3c.json file in repos (and then from the W3C API) otherwise.

Closes #32

Draft PR because README still needs to be updated.

### CODE REVIEW

File: src/build-index.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ const determineFilename = require("./determine-filename.js");
 const determineTestPath = require("./determine-testpath.js");
 const extractPages = require("./extract-pages.js");
 const fetchInfo = require("./fetch-info.js");
+const fetchGroups = require("./fetch-groups.js");
```
Comment (dontcallmedom) [inline src/build-index.js:21]: that file is missing :)

File: src/fetch-groups.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    return body;
+  }
+
+  const w3cOptions = options?.w3cApiKey ? { headers: {
+    Authorization: `W3C-API apikey="${options.w3cApiKey}"`
+  }} : {};
+
+  for (const spec of specs) {
+    // Note: same RegExp logic to detect WHATWG, TC39, SVG, Khronos Group and
+    // GitHub URLs as in compute-repository.
```
Comment (dontcallmedom) [inline src/fetch-groups.js:51]: is there an opportunity to factor that logic out to reduce the maintenance burden?

File: src/fetch-groups.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    const whatwg = spec.url.match(/^https:\/\/([^\.]*).spec.whatwg.org\//);
+    if (whatwg) {
+      spec.organization = spec.organization ?? "WHATWG";
+      spec.groups = spec.groups ?? [{
+        name: "WHATWG",
```
Comment (dontcallmedom) [inline src/fetch-groups.js:56]: the WHATWG has a notion of [workstreams](https://whatwg.org/workstream-policy#workstream) - workstreams have “participants”, so maybe they're the closest thing to a notion of group in that context.

File: src/fetch-groups.js [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+        url: "https://tc39.es/"
+      }];
+      continue;
+    }
+
+    const webgl = spec.url.match(/^https:\/\/www\.khronos\.org\/registry\/webgl\//);
+    if (webgl) {
+      spec.organization = spec.organization ?? "Khronos Group";
+      spec.groups = spec.groups ?? [{
+        name: "WebGL",
```
Comment (dontcallmedom) [inline src/fetch-groups.js:76]:
```suggestion
        name: "WebGL Working Group",
```

File: index.json [hunk 5]
```diff
@@ -4424,6 +5677,13 @@
       "nightlyUrl": "https://w3c.github.io/accname/"
     },
     "seriesVersion": "1.2",
+    "organization": "W3C",
+    "groups": [
+      {
+        "name": "Devices and Sensors Working Group",
```
Comment (dontcallmedom) [inline index.json:5683]: This is wrong, so there must be a bug somewhere :)

(222 specs are assigned to DAS)

File: src/fetch-groups.js (hunk 3 above)
Comment (dontcallmedom) [reply src/fetch-groups.js:56]: The [list of workstreams](https://whatwg.org/workstreams) is generated from a [WHATWG-SG maintained JSON file](https://github.com/whatwg/sg/blob/main/db.json)

File: index.json (hunk 5 above)
Comment (tidoust) [reply index.json:5683]: Such a productive working group...

Comment (dontcallmedom) [reply index.json]: for reference, here is the outlook with the fixed code
```
    6 "Accessible Rich Internet Applications Working Group"
      2 "Audio Working Group"
      2 "Browser Testing and Tools Working Group"
    113 "Cascading Style Sheets (CSS) Working Group"
     15 "Devices and Sensors Working Group"
      1 "Distributed Tracing Working Group"
      1 "Geolocation Working Group"
      2 "GPU for the Web Working Group"
      2 "HTML Media Extensions Working Group"
      2 "Immersive Web Community Group"
      8 "Immersive Web Working Group"
      1 "Internationalization Working Group"
      1 "Math Working Group"
      5 "Media Working Group"
      1 "Pointer Events Working Group"
      2 "Privacy Community Group"
      3 "Second Screen Working Group"
      1 "Service Workers Working Group"
      9 "SVG Working Group"
     17 "TC39"
      2 "Technical Architecture Group"
      1 "Timed Text Working Group"
      1 "Tracking Protection Working Group"
     14 "Web Application Security Working Group"
     19 "Web Applications Working Group"
      3 "WebAssembly Working Group"
      1 "Web Authentication Working Group"
      1 "Web Bluetooth Community Group"
      1 "Web Cryptography Working Group"
      1 "Web Events Working Group"
      1 "Web Fonts Working Group"
     44 "WebGL"
      1 "Web Machine Learning Working Group"
      1 "Web NFC Community Group"
      5 "Web Payments Working Group"
     18 "Web Performance Working Group"
     64 "Web Platform Incubator Community Group"
      4 "Web Platform Working Group"
     16 "Web Real-Time Communications Working Group"
      1 "WebTransport Working Group"
     14 "WHATWG"
```
based on `jq '.[].groups[].name' index.json|sort|uniq -c`

File: src/fetch-groups.js (hunk 3 above)
Comment (tidoust) [reply src/fetch-groups.js:56]: So, it's easy to retrieve the name of the workstream but there are no dedicated URL per workstream as far as I can tell. If we want the `url` property to act as a sort of group identifier, the best is probably to link to the spec itself for now (today, there's a 1-to-1 mapping between workstreams and specs). Any other idea?

Comment (dontcallmedom) [reply src/fetch-groups.js:56]: no other idea, and +1 to using the spec itself for now given the congruence

File: src/build-index.js [hunk 6]
```diff
@@ -56,6 +56,15 @@ async function sleep(ms) {
   return new Promise(resolve => setTimeout(resolve, ms));
 }
 
+// Log function that can be inserted in a "then" chain
+function dolog(msg) {
+  return arg => {
+    console.log(msg);
```
Comment (dontcallmedom) [inline src/build-index.js:62]: no cute emoji?!?

File: src/fetch-groups.js (hunk 2 above)
Comment (tidoust) [reply src/fetch-groups.js]: Done in latest commit.

File: src/build-index.js (hunk 6 above)
Comment (tidoust) [reply src/build-index.js:62]: Not a single one. Browser-specs is serious business.

**Comment by tidoust** [comment]
I'm a bit at a loss as to why CI fails. It complains that the `organization` property is missing on one of the specs but, as far as I can tell, all specs have that property in the version of `index.json` I pushed with the PR, tests pass locally (and I didn't forget any file for once).

**Review by dontcallmedom — APPROVED** [review]
I don't understand nor can reproduce the CI failure either 🤷

File: src/build-index.js (hunk 6 above)
Comment (dontcallmedom) [reply src/build-index.js:62]: 😭

**Comment by dontcallmedom** [comment]
maybe bump the package minor version while we're at it? (new feature)

**Comment by dontcallmedom** [comment]
hmm.. or is a schema change a major version bump?

**Comment by tidoust** [comment]
Our [versioning rules](https://github.com/w3c/browser-specs#versioning) suggest that a major version bump is not needed since this does not introduce a backwards-incompatible change.

Note that version bumps are done when I release a new version (for instance, I just released a [v1.42.0](https://github.com/w3c/browser-specs/commit/779a01800e322013c728e2e29a9475674365359b) with the new specs), so not something that need to be done in the PR.

**Comment by tidoust** [comment]
Merging with master solved the CI issue. No idea why...

**Pull request merged**